tests/test_hy2py.py

```diff
@@ -0,0 +1,120 @@
+
+    assert getattr(m, mangle(u"identifier-that-has☝️💯☝️-to-be-mangled")) == "ponies"
+
+    assert m.mynumber == 3
+    assert m.myhex == 0x123
+    assert m.mylong - 1234567890987654321234567890987654320 == 1
+    assert m.myfloat == 3.34e15
+    assert math.isnan(m.mynan)
+    assert math.isinf(m.pinf)
+    assert m.pinf > 0
+    assert math.isinf(m.ninf)
+    assert m.ninf < 0
+
+    assert m.num_expr == 9
+
+    assert m.mylist == [1, 2, 3]
+    assert m.mytuple == ("a", "b", "c")
+    assert m.myset == {4, 5, 6}
+    assert m.mydict == {7: 8, 9: 900, 10: 15}
+
+    assert m.mylistcomp == [1, 3, 5, 7, 9]
+    assert m.mysetcomp == {0, 2, 4}
+    assert m.mydictcomp == dict(a="A", b="B", d="D", e="E")
+    assert type(m.mygenexpr) is type( (x for x in [1, 2, 3]) )
+    assert list(itertools.islice(m.mygenexpr, 5)) == [1, 3, 1, 3, 1]
+
+    assert m.attr_ref is str.upper
+    assert m.subscript == "l"
+    assert m.myslice == "el"
+    assert m.call == 5
+    assert m.comparison is True
+    assert m.boolexpr is True
+    assert m.condexpr == "y"
+    assert type(m.mylambda) is type(lambda x: x + "z")
+    assert m.mylambda("a") == "az"
+
+    assert m.augassign == 25
+
+    assert m.delstatement == ["a", "c", "d", "e"]
+
+    assert m.math is math
+    assert m.sqrt is math.sqrt
+    assert m.sine is math.sin
+    import datetime
+    assert m.timedelta is datetime.timedelta
+
+    assert m.if_block == "cd"
+    assert m.while_block == "xxxxe"
+    assert m.cont_and_break == "xyzxyzxxyzxy"
+    assert m.for_block == "fufifo"
+    assert m.caught_assertion is True
+    assert m.ran_finally is True
+    assert m.myraise == "payload"
+    assert m.ran_try_else is True
+
+    assert type(m.fun) is type(lambda x: x)
+    if not astor_issue101_workaround:
+        assert m.fun.__doc__ == "function docstring"
+    assert m.funcall1 == [
+        1, 2, 3, 4, ("a", "b", "c"), [("k1", "v1"), ("k2", "v2")]]
+    assert m.funcall2 == [
+        7, 8, 9, 10, (11,), [("x1", "y1"), ("x2", "y2")]]
+
+    assert m.myret == 1
+    assert m.myyield == ["a", "b", "c"]
+    assert m.mydecorated.newattr == "hello"
+    assert m.myglobal == 103
+
+    class C: pass
+    assert type(m.C1) is type(C)
+
+    if not astor_issue101_workaround:
+        assert m.C2.__doc__ == "class docstring"
+    assert issubclass(m.C2, m.C1)
+    assert (m.C2.attr1, m.C2.attr2, m.C2.attr3) == (5, 6, 7)
+
+    assert m.closed == ["v2", "v1"]
+
+
+def import_from_path(name, path):
+    if PY36:
+        import importlib.util
+        spec = importlib.util.spec_from_file_location(name, path)
+        m = importlib.util.module_from_spec(spec)
+        spec.loader.exec_module(m)
```

Comment:
This works on PY35 as well, for what its worth. `spec_from_file_location` came with 3.4 and `module_from_spec` with 3.5

Reply:
I tried enabling this for Python 3.5 and I got a weird crash inside pytest, so I've changed it back.

Reply:
Yeah, probably won't work until #1518 is in. I tested it there. We don't implement the importlib APIs correctly.